### PR TITLE
BUG_FIX: closing a SpectrumLine was not removing it from SpectrumFigure

### PR DESCRIPTION
**Pull request by francisco-dlp**
This fixes half of #527 and possibly other unreported issues.

### CODE REVIEW

File: hyperspy/drawing/spectrum.py [hunk 1]
```diff
@@ -339,6 +341,8 @@ def close(self):
         if self.text and self.text in self.ax.texts:
             self.ax.texts.remove(self.text)
         self.axes_manager.disconnect(self.update)
+        if self.spectrum_figure:
+            self.spectrum_figure.ax_lines.remove(self)
```
Comment (vidartf) [inline hyperspy/drawing/spectrum.py]: Do we need to check/remove from `self.right_ax_line` as well?

Comment (francisco-dlp) [inline hyperspy/drawing/spectrum.py]: Thank you for reporting the issue. I have just committed a fix.

I know that the fix is ugly. Hopefully we'll refactor the drawing code once #423 is merged.

File: hyperspy/drawing/spectrum.py [hunk 2]
```diff
@@ -81,9 +80,11 @@ def add_line(self, line, ax='left'):
             if line.axes_manager is None:
                 line.axes_manager = self.axes_manager
             self.ax_lines.append(line)
+            line.sf_lines = self.ax_lines
         elif ax == 'right':
             line.ax = self.right_ax
             self.right_ax_lines.append(line)
+            line.sf_lines = self.ax_lines
```
Comment (vidartf) [inline hyperspy/drawing/spectrum.py]: This should be `line.sf_lines = self.right_ax_lines`? :)

**Comment by francisco-dlp** [comment]
With a bit of luck this should be ready now, is it @vidartf?

**Comment by vidartf** [comment]
@francisco-dlp All good :)

**Pull request merged**